src/fs.cpp

```diff
@@ -12,4 +12,18 @@ FILE *freopen(const fs::path& p, const char *mode, FILE *stream)
     return ::freopen(p.string().c_str(), mode, stream);
 }
 
+// Create and return temporary subdirectory for test outputs
+fs::path unit_test_directory(){
+    std::string sub_dir = "bitcoin_unit_tests";
+    fs::path dir(fs::temp_directory_path() / sub_dir);
+
+    if(fs::create_directory(dir) || fs::exists(dir))
+    {
+        return dir;
+    }
+    else{
```

Comment:
Don't forget to install `clang-format` and run the https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy script to preempt whitespace nitpicking.